framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -1355,6 +1389,107 @@ bool VulkanReplayConsumerBase::GetOverrideDevice(InstanceInfo* instance_info, Ph
     return true;
 }
 
+bool VulkanReplayConsumerBase::GetOverrideDeviceGroup(InstanceInfo*                  instance_info,
+                                                      PhysicalDeviceInfo*            physical_device_info,
+                                                      VkDeviceGroupDeviceCreateInfo& device_group_create_info,
+                                                      std::vector<VkPhysicalDevice>& replay_device_group)
+{
+    VkInstance instance               = instance_info->handle;
+    auto       table                  = GetInstanceTable(instance);
+    uint32_t   replay_gpu_group_count = 0;
+    VkResult   err = table->EnumeratePhysicalDeviceGroups(instance, &replay_gpu_group_count, nullptr);
+    GFXRECON_ASSERT(!err);
+
+    std::vector<VkPhysicalDeviceGroupProperties> replay_group_propss;
+    if (replay_gpu_group_count)
+    {
+        replay_group_propss.resize(replay_gpu_group_count);
+        err = table->EnumeratePhysicalDeviceGroups(instance, &replay_gpu_group_count, replay_group_propss.data());
+        GFXRECON_ASSERT(!err);
+    }
+    else
+    {
+        return false;
+    }
+
+    // Check for a valid override device index.
+    if (options_.override_gpu_group_index >= static_cast<int32_t>(replay_gpu_group_count))
+    {
+        GFXRECON_LOG_ERROR(
+            "The zero-based index specified for replay device override (%d) exceeds the total number of "
+            "available physical device groups (%d). The specified index requires that at least %d devices be "
+            "available. The override will not be applied.",
+            options_.override_gpu_group_index,
+            replay_gpu_group_count,
+            (options_.override_gpu_group_index + 1));
+        return false;
+    }
+
+    const auto& override_group_props = replay_group_propss[options_.override_gpu_group_index];
+    replay_device_group.clear();
+    std::copy(override_group_props.physicalDevices,
+              override_group_props.physicalDevices + override_group_props.physicalDeviceCount,
+              std::back_inserter(replay_device_group));
+
+    device_group_create_info.physicalDeviceCount = override_group_props.physicalDeviceCount;
+    device_group_create_info.pPhysicalDevices    = replay_device_group.data();
+
+    GFXRECON_LOG_INFO("Creating logical device from manually specified GPU group[%d]",
+                      options_.override_gpu_group_index);
+    GFXRECON_LOG_INFO("  Available device groups are:");
+
+    std::string override_device_name;
+    bool        first_override_device_name = false;
+
+    for (uint32_t i = 0; i < replay_gpu_group_count; ++i)
+    {
+        GFXRECON_LOG_INFO("group[%d]:", i);
```

Comment:
I think you forgot two indentation spaces here in the LOG_INFO text: "  group[%d]:"

Reply:
Thanks. Fixed.